etc/kayobe/kolla.yml

```diff
@@ -352,18 +352,22 @@ kolla_build_blocks:
     {% if stackhpc_kolla_clean_up_repo_mirrors | bool %}
     {% if kolla_base_distro in ['centos', 'rocky'] %}
     RUN \
-        tar -xzf /etc/yum.repos.d.backup/repos.tar.gz -C /etc/yum.repos.d && \
-        if grep -r '{{ stackhpc_repo_mirror_url }}' /etc/yum.repos.d; then \
-           echo "Found repository mirror in Yum repositories"; \
-           exit 1; \
-        fi && \
-        rm -rf /etc/yum.repos.d.backup/
+        if [ -f /etc/yum.repos.d.backup/repos.tar.gz ]; then \
+          tar -xzf /etc/yum.repos.d.backup/repos.tar.gz -C /etc/yum.repos.d && \
+          if grep -r '{{ stackhpc_repo_mirror_url }}' /etc/yum.repos.d; then \
+            echo "Found repository mirror in Yum repositories"; \
+            exit 1; \
+          fi && \
+          rm -rf /etc/yum.repos.d.backup/
+        fi
     {% else %}
     RUN \
-        mv /etc/apt/sources.list.backup /etc/apt/sources.list && \
-        if grep -r '{{ stackhpc_repo_mirror_url }}' /etc/apt/sources.list; then \
-           echo "Found repository mirror in APT repositories"; \
-           exit 1; \
+        if [ -f /etc/apt/sources.list.backup ]; then \
+          mv /etc/apt/sources.list.backup /etc/apt/sources.list && \
+          if grep -r '{{ stackhpc_repo_mirror_url }}' /etc/apt/sources.list; then \
```

Comment:
Possibly move the check outside the conditional so that we still fail if the stackhpc mirrors are configured?

Reply:
It should still fail when the parent container image is built, as the footer would have ran he before.

Reply:
Sorry, I meant this check:
```
          if grep -r '{{ stackhpc_repo_mirror_url }}' /etc/yum.repos.d; then \
            echo "Found repository mirror in Yum repositories"; \
            exit 1; \
          fi && \
```

Reply:
Done